### ISSUE TITLE
Feature: Support 'Test Connection' action in connection dialog

### ISSUE THREAD

**Issue by cheenamalhotra** [microsoft/azuredatastudio#23419]
There have been multiple requests for the same from customers, so opening this feature request to track activity and gather feedback.

**Proposal:**
Support new action: "Test Connection" as below:

<img src="https://github.com/microsoft/azuredatastudio/assets/13396919/6b0cc0c8-1c60-404b-97af-06e97bd70a22" width=500 />

**Actions:**
* When clicked, a temporary connection is made, result is displayed to user, and error dialog pops up in case of error: 

   <img src="https://github.com/microsoft/azuredatastudio/assets/13396919/14ad6425-56c5-4548-b846-c62c6bd22ab7" width=500 />
  <img src="https://github.com/microsoft/azuredatastudio/assets/13396919/00b470ac-912f-4931-b3e8-d5216750bdd3" width=500 />
  
_(Please consult design team for accessibility friendly color scheme)_
 
Feel free to post additional thoughts to refine feature request if needed.

**Comment by erinstellato-ms** [comment]
@cheenamalhotra if the connection fails, do we surface any errors to help the user understand why it fails?

Trying to make sure I'm clear on when I would use Test Connection versus just entering everything and selecting Connect.

**Comment by cheenamalhotra** [comment]
Yes, the action would cover showing error message dialog as it would when user selects 'Connect'.
